### PR TITLE
Add react-native support

### PR DESCRIPTION
**Pull request by yhan64**
image

### CODE REVIEW

File: index.native.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import Breakdancer from './';
+import { Dimensions } from 'react-native';
+
+export default class NativeDancer extends Breakdancer {
+  constructor (specification) {
```
Comment (3rd-Eden) [inline index.native.js]: You can remove the constructor definition here.

File: index.native.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  /**
+   * Return the current view port.
+   *
+   * @returns {Object} viewport
+   * @public
+   */
+  viewport () {
+    return Dimensions.get('window');
+  }
+
```
Comment (3rd-Eden) [inline index.native.js]: Remove extra white space, so there's only 1 empty line instead of 2

File: package.json [hunk 3]
```diff
@@ -40,12 +45,17 @@
   },
   "dependencies": {
     "babel-preset-es2015": "6.9.x",
+    "babel-preset-react": "^6.23.0",
```
Comment (3rd-Eden) [inline package.json]: Is this really needed as dependency as no React/JSX is used in the codebase

File: index.js [hunk 4]
```diff
@@ -41,6 +41,9 @@ export default class Breakdancer {
    * @private
    */
   normalize (specification) {
+    if (!specification) {
```
Comment (3rd-Eden) [inline index.js]: I don't really see the need of this change. If there is no specification of breakpoints there would be absolutely no need to use this module in the first place.

File: index.native.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  /**
+   * Lookup the view port width.
+   *
+   * @returns {Number} Current width.
+   * @public
+   */
+  width () {
+    return Dimensions.get('window').width;
```
Comment (Swaagie) [inline index.native.js]: really minor, code could be DRY-up by calling `return this.viewport().width;`

File: index.native.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  /**
+   * Lookup the view port height.
+   *
+   * @returns {Number} Current height.
+   * @public
+   */
+  height () {
+    return Dimensions.get('window').height;
```
Comment (Swaagie) [inline index.native.js]: same as `width`

File: index.native.js (hunk 5 above)
Comment (3rd-Eden) [inline index.native.js]: But that would introduce an extra function call as overhead, which is not something you really want.

**Comment by 3rd-Eden** [comment]
Published as 1.1.0

**Comment by fabiomcosta** [comment]
Now to use the latest version of `breakdancer` on a project I'll also need  to install`react-native`, even on web projects.
Was that intended?
Maybe `react-native` should be an optional dependency or just not be listed on the package.json file.

**Comment by fabiomcosta** [comment]
Another issue is that `react-native` has a `react` peer dependency of version `0.16.x` https://github.com/facebook/react-native/blob/v0.46.1/package.json#L136 and react `0.15.x` was added as `peerDependency` on this package.

**Pull request merged**